manticore/platforms/evm.py

```diff
@@ -1968,7 +2012,11 @@ def create_account(self, address=None, balance=0, code='', storage=None, nonce=0
         return address
 
     def create_contract(self, price=0, address=None, caller=None, balance=0, init=None, gas=2300):
-        '''
+        ''' Create a contract account. Sends a transaction to initialize the contract
```

Comment:
~Does this even work with `caller = None`? ~ 
Caller must be defined in accounts. So I guess this is always called with a caller. (https://github.com/trailofbits/manticore/blob/2a4d62a2cc634a27c452841f420eb4e329532c32/manticore/platforms/evm.py#L2245-L2246)

So the question is why to have an `address` parameter if it is deterministic calculated using the caller.

Reply:
I am not sure about the rest of the downstream function calls, but `create_contract` itself should be fine if `caller` is `None`. Anyway, we have to keep `caller=None` in the function signature at this point because we have default arguments before, so the only way to remove the default value from `caller` would be to move its argument position earlier, which would break the API for any callers that call it with positional arguments.